backend/src/v2/driver/driver_test.go

```diff
@@ -635,3 +635,56 @@ func Test_initPodSpecPatch_resourceRequests(t *testing.T) {
 		})
 	}
 }
+
+func Test_initPodSpecPatch_inputTaskFinalStatus(t *testing.T) {
+	finalStatusStruct, err := structpb.NewStruct(map[string]interface{}{
+		"state":                   "test-state",
+		"pipelineTaskName":        "test-pipeline-task-name",
+		"pipelineJobResourceName": "test-job-resource-name",
+		"error":                   map[string]interface{}{},
+	})
+
+	containerSpec := &pipelinespec.PipelineDeploymentConfig_PipelineContainerSpec{
+		Image:   "python:3.9",
+		Command: []string{"sh", "-ec", "python3 -m kfp.components.executor_main"},
+		Args:    []string{"--executor-input", "{{$}}", "--function_to_execute", "exit-op"},
+	}
+
+	componentSpec := &pipelinespec.ComponentSpec{
+		Implementation: &pipelinespec.ComponentSpec_ExecutorLabel{ExecutorLabel: "exec-exit-op"},
+		InputDefinitions: &pipelinespec.ComponentInputsSpec{
+			Parameters: map[string]*pipelinespec.ComponentInputsSpec_ParameterSpec{
+				"status": {ParameterType: pipelinespec.ParameterType_TASK_FINAL_STATUS},
+			},
+		},
+	}
+
+	executorInput := &pipelinespec.ExecutorInput{
+		Inputs: &pipelinespec.ExecutorInput_Inputs{
+			ParameterValues: map[string]*structpb.Value{
+				"status": {
+					Kind: &structpb.Value_StructValue{
+						StructValue: finalStatusStruct,
+					},
+				},
+			},
+		},
+	}
+
+	podSpec, err := initPodSpecPatch(
+		containerSpec,
+		componentSpec,
+		executorInput,
+		27,
+		"test",
+		"0254beba-0be4-4065-8d97-7dc5e3adf300",
+		"1",
+		"false",
+		"false",
+	)
+
+	podSpecString, err := json.Marshal(podSpec)
+	assert.Nil(t, err)
+	assert.Contains(t, string(podSpecString), "--executor_input\",\"{\\\"inputs\\\":{\\\"parameterValues\\\":{\\\"status\\\":{\\\"error\\\":{}, \\\"pipeline_job_resource_name\\\":\\\"test-job-resource-name\\\", \\\"pipeline_task_name\\\":\\\"test-pipeline-task-name\\\", \\\"state\\\":\\\"test-state\\\"}}}}")
```

Comment:
The test failure is because the keys are now camel case. There's also some spacing issues.

My suggestion here is since `podSpec` is an object, you can just parse the `Command` field in the container directly. I'd also suggest parsing the JSON to a map[string]interface{} since you are relying that JSON marshaling will always return the keys in order by doing a JSON string comparison.